`HugeResourceTest.testUploadDelayed()` fails when `RetainedContentLoaderFactory`'s `reject` is not explicitly disabled

**Jetty version(s)**
12.1.x

**Jetty Environment**
ee10, ee11

**Description**
The test currently passes because a `RetainedContentLoaderFactory` with the `reject` parameter explicitly set to false is passed to the `EagerContentHandler`'s constructor.

Otherwise, `HugeResourceTest.testUploadDelayed()` fails with HTTP/413 Payload Too Large when the `EagerContentHandler`'s default constructor is used:

```
java.lang.AssertionError: HTTP Response Code
Expected: is <200>
     but: was <413>
```

Note: The test currently passes because `EagerContentHandler` is configured with a `RetainedContentLoaderFactory` that has `reject` explicitly set to false.

--- a/jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/handler/EagerContentHandler.java
+++ b/jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/handler/EagerContentHandler.java
@@ -437,18 +437,25 @@ public class EagerContentHandler extends ConditionalHandler.ElseNext
         private final int _framingOverhead;
         private final boolean _reject;
 
+        /**
+         * Creates a {@code RetainedContentLoaderFactory} with heuristically determined values for the maximum number
+         * bytes to retain and the number of bytes to include in the estimated size per {@link Content.Chunk} to allow
+         * for framing overheads in the transport and no rejection of the request when the maximum number bytes to
+         * retain is exceeded.
+         */
         public RetainedContentLoaderFactory()
         {
-            this(-1, -1, true);
+            this(-1, -1, false);
         }
 
         /**
+         * Creates a {@code RetainedContentLoaderFactory}.
          * @param maxRetainedBytes the maximum number bytes to retain whilst eagerly loading, which
          *                         includes the content bytes and any {@code framingOverhead} per chunk;
          *                         or -1 for a heuristically determined value that will not increase memory commitment.
          * @param framingOverhead the number of bytes to include in the estimated size per {@link Content.Chunk} to allow
          *                      for framing overheads in the transport. Since the content is retained rather than copied, any
-         *                      framing data is also retained in the IO buffer.
+         *                      framing data is also retained in the IO buffer; or -1 for a heuristically determined value.
          * @param reject if {@code true}, then if {@code maxRetainBytes} is exceeded, the request is rejected with a
          *               {@link HttpStatus#PAYLOAD_TOO_LARGE_413} response.
          */

--- a/jetty-ee10/jetty-ee10-webapp/src/test/java/org/eclipse/jetty/ee10/webapp/HugeResourceTest.java
+++ b/jetty-ee10/jetty-ee10-webapp/src/test/java/org/eclipse/jetty/ee10/webapp/HugeResourceTest.java
@@ -219,11 +219,7 @@ public class HugeResourceTest
         ServletHolder holder = context.addServlet(MultipartServlet.class, "/multipart");
         holder.getRegistration().setMultipartConfig(multipartConfig);
 
-        EagerContentHandler eagerContentHandler = new EagerContentHandler(
-            new EagerContentHandler.FormContentLoaderFactory(),
-            new EagerContentHandler.MultiPartContentLoaderFactory(),
-            new EagerContentHandler.RetainedContentLoaderFactory(-1, -1, false)
-        );
+        EagerContentHandler eagerContentHandler = new EagerContentHandler();
         server.setHandler(eagerContentHandler);
         httpConfig.setDelayDispatchUntilContent(false);
 

--- a/jetty-ee11/jetty-ee11-webapp/src/test/java/org/eclipse/jetty/ee11/webapp/HugeResourceTest.java
+++ b/jetty-ee11/jetty-ee11-webapp/src/test/java/org/eclipse/jetty/ee11/webapp/HugeResourceTest.java
@@ -219,11 +219,7 @@ public class HugeResourceTest
         ServletHolder holder = context.addServlet(MultipartServlet.class, "/multipart");
         holder.getRegistration().setMultipartConfig(multipartConfig);
 
-        EagerContentHandler eagerContentHandler = new EagerContentHandler(
-            new EagerContentHandler.FormContentLoaderFactory(),
-            new EagerContentHandler.MultiPartContentLoaderFactory(),
-            new EagerContentHandler.RetainedContentLoaderFactory(-1, -1, false)
-        );
+        EagerContentHandler eagerContentHandler = new EagerContentHandler();
         server.setHandler(eagerContentHandler);
         httpConfig.setDelayDispatchUntilContent(false);
 

